lib/t_web/socket/user_socket.ex

```diff
@@ -0,0 +1,270 @@
+    case Accounts.submit_profile(user_id, params) do
+      {:ok, profile} -> {:ok, render_profile(profile), assigns}
+      {:error, %Ecto.Changeset{} = cs} -> {:error, error(:changeset, cs), assigns}
+    end
+  end
+
+  def handle_event("set-audio-only", %{"bool" => bool}, %{user_id: user_id} = assigns) do
+    Accounts.set_audio_only(user_id, bool)
+    {:ok, assigns}
+  end
+
+  # /PROFILE
+
+  # FEED
+
+  def handle_event("start-feed", params, assigns) do
+  end
+
+  def handle_event("more-feed", params, assigns) do
+  end
+
+  def handle_event("seen", %{"user_id" => user_id}, assigns) do
+    Feeds.mark_profile_seen(user_id, by: assigns.user_id)
+    {:ok, assigns}
+  end
+
+  # /FEED
+
+  # LIKES
+
+  def handle_event("like", %{"user_id" => user_id}, assigns) do
+    case Matches.like_user(assigns.user_id, user_id) do
+      {:ok, %{match: _no_match = nil}} ->
+        {:ok, assigns}
+
+      {:ok, %{match: match, audio_only: audio_only}} ->
+        {:ok, %{match_id: match.id, audio_only: audio_only}, assigns}
+
+      {:error, _step, _reason, _changes} ->
+        {:ok, assigns}
+    end
+  end
+
+  def handle_event("decline", %{"user_id" => user_id}, assigns) do
+    Matches.decline_like(assigns.user_id, user_id)
+    {:ok, assigns}
+  end
+
+  # /LIKES
+
+  # MATCHES
+
+  def handle_event("archive-match", %{"match_id" => match_id}, assigns) do
+    Matches.mark_match_archived(match_id, assigns.user_id)
+    {:ok, assigns}
+  end
+
+  def handle_event("unarchive-match", %{"match_id" => match_id}, assigns) do
+    Matches.unmatch_match(match_id, assigns.user_id)
+    {:ok, assigns}
+  end
+
+  def handle_event("seen", %{"expired_match_id" => match_id}, assigns) do
+    Matches.delete_expired_match(match_id, assigns.user_id)
+    {:ok, assigns}
+  end
+
+  def handle_event("unmatch", %{"user_id" => user_id}, assigns) do
+    {:ok, assigns}
+  end
+
+  # /MATCHES
+
+  # TIMESLOTS
+
+  def handle_event("offer-slots", %{"slots" => slots, "user_id" => user_id}, assigns) do
+    case Matches.save_slots_offer_for_user(assigns.user_id, user_id, slots) do
+      {:ok, _timeslot} ->
+        {:ok, assigns}
+
+      {:error, %Ecto.Changeset{} = changeset} ->
+        {:error, error(:changeset, changeset), assigns}
+    end
+  end
+
+  def handle_event("pick-slot", %{"slot" => slot, "user_id" => user_id}, assigns) do
+  end
+
+  def handle_event("cancel-slot", %{"user_id" => user_id}, assigns) do
+  end
+
+  # /TIMESLOTS
+
+  # CONTACTS
+  # /CONTACTS
+
+  # CALLS
+
+  def handle_event("call", %{"user_id" => user_id}, assigns) do
+    case Calls.call(assigns.user_id, user_id, utc_now(assigns)) do
+      {:ok, call_id} ->
+        {:ok, %{"call_id" => call_id, "ice-servers" => [], "call_topics" => []}, assigns}
+
+      {:error, reason} ->
+        {:error, error(reason), assigns}
+    end
+  end
+
+  def handle_event("join-call", %{"user_id" => user_id}, assigns) do
+    {:ok, %{"caller" => %{}, "ice-servers" => [], "call_topics" => []}, assigns}
```

Comment:
actually we don't need join-call, we can return this info on `pick-up`

Reply:
this doesn't account for the case when `called` picks-up after `caller` hung-up, or rather it does but not in the best way:

```
1. caller calls
2. caller hangs up
3. called gets notification, connects to socket, picks up
4. pick up fails since the call has ended, called user is confused
```

Reply:
called user probably still needs to ask backend if the caller is still calling right after connecting to socket

```
1. caller calls
2. caller hangs up
3. called gets notification, connects to socket, pushes something like [1,"connect-call",{"call_id":"234"}]
4. receives `[1,"error",[404,"call has ended"]]`
5. this happens very quickly and the user doesn't notice there's been a call
```

Reply:
can be called `check-call`, `call-alive`, `connect-call`, `join-call` ...